tests/framework/extensions/clusters/clusters.go

```diff
@@ -383,6 +383,62 @@ func NewK3SRKE2ClusterConfig(clusterName, namespace string, clustersConfig *Clus
 		RKEClusterSpecCommon: rkeSpecCommon,
 		MachinePools:         machinePools,
 	}
+
+	if clustersConfig.CloudProvider == "aws" {
+		rkeConfig.MachineGlobalConfig = rkev1.GenericMap{
+			Data: map[string]interface{}{
+				"cloud-provider-name": "aws",
+			},
+		}
+		rkeConfig.MachineSelectorConfig = []rkev1.RKESystemConfig{
+			{
+				Config: rkev1.GenericMap{
+					Data: map[string]interface{}{
+						"kubelet-arg": []string{
+							"cloud-provider=external",
+						},
+					},
+				},
+				MachineLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"rke.cattle.io/etcd-role": "true",
+					},
+				},
+			},
+			{
+				Config: rkev1.GenericMap{
+					Data: map[string]interface{}{
+						"kubeapi-server-arg": []string{
+							"cloud-provider=external",
+						},
+						"kube-controller-manager-arg": []string{
+							"cloud-provider=external",
+						},
+					},
+				},
+				MachineLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"rke.cattle.io/control-plane-role": "true",
+					},
+				},
+			},
+			{
+				Config: rkev1.GenericMap{
+					Data: map[string]interface{}{
+						"kubelet-arg": []string{
+							"cloud-provider=external",
+						},
+					},
+				},
+				MachineLabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"rke.cattle.io/worker-role": "true",
+					},
+				},
+			},
```

Comment:
I think we can create multiple constructors here, just like how our workloads templates are working:

1. A public RKESystem constructor
2. A potential private Data map[string]any constructor, for each given key it can hardcode `"cloud-provider=external"` values.
3. A private RKESystem constructor i.e. with the following Data field hardcoded, which uses (1) and (2):
```					
					Data: map[string]interface{}{
						"kubelet-arg": []string{
							"cloud-provider=external",
						},
					},
```
Also, I think some of the strings are used pretty much in every object, i.e. "cloud-provider=external". I also think we can have; worker role labels, and etcd role labels as consts or enums too.

Of course, everything also depends on how the other providers align with these suggestions. Would others be similar to this?

Reply:
agreed. It would look much cleaner.

Reply:
I believe this is resolved, lmk what you think about the new implementation